src/hardware/uni-t-ut8802e/protocol.c

```diff
@@ -0,0 +1,430 @@
+
+    sr_sw_limits_update_samples_read(&devc->limits, 1);
+    if (sr_sw_limits_check(&devc->limits))
+        sr_dev_acquisition_stop(sdi);
+
+    if (feedbuff.analog.meaning)
+        g_slist_free(feedbuff.analog.meaning->channels);
+
+    return SR_OK;
+}
+
+static int process_buffer(struct sr_dev_inst *sdi) {
+    struct dev_context *devc;
+    uint8_t *pkt;
+    uint8_t v8;
+    size_t pkt_len, remain, idx;
+    int ret;
+    GString *spew;
+
+    devc = sdi->priv;
+    pkt = &devc->packet[0];
+
+    do {
+        if (devc->packet_len < 1 * sizeof(uint8_t))
+            return SR_OK;
+
+        v8 = R8(&pkt[0]);
+        if (v8 != FRAME_MAGIC)
+            break;
+
+        pkt_len = 8;
+        sr_spew("Got a expected packet lenght %zu, have %zu",
+                pkt_len, devc->packet_len);
+
+        if (pkt_len > devc->packet_len)
+            return SR_OK;
+
+        spew = sr_hexdump_new(pkt, pkt_len);
+        sr_spew("Packet to procceses, len %zu, bytes: %s", pkt_len, spew->str);
+        sr_hexdump_free(spew);
+
+        ret = process_packet(sdi, pkt, pkt_len);
+        if (ret == SR_ERR_DATA) {
+            /* Verification failed */
+            break;
+        }
+
+        remain = devc->packet_len - pkt_len;
+        if (remain)
+            memmove(&pkt[0], &pkt[pkt_len], remain);
+        devc->packet_len -= pkt_len;
+    } while (1);
+
+    if (devc->packet_len < 1 * sizeof(uint8_t))
+        return SR_OK;
+
+    for (idx = 1; idx < devc->packet_len; idx++) {
+        if (devc->packet_len - idx < sizeof(uint8_t)) {
+            pkt[0] = pkt[idx];
+            devc->packet_len = 1;
+            return SR_OK;
+        }
+
+        v8 = R8(&pkt[idx]);
+        if (v8 != FRAME_MAGIC)
+            continue;
+
+        remain = devc->packet_len - idx;
+        if (remain)
+            memmove(&pkt[0], &pkt[idx], remain);
+        devc->packet_len -= idx;
+        break;
+    }
+    return SR_OK;
+}
+
+static int ut8802e_receive_data(struct sr_dev_inst *sdi) {
+    struct dev_context *devc;
+    struct sr_serial_dev_inst *serial;
+    size_t len;
+    uint8_t *data;
+
+    devc = sdi->priv;
+    serial = sdi->conn;
+
+    if (devc->packet_len == sizeof(devc->packet)) {
+        (void)process_packet(sdi, &devc->packet[0], devc->packet_len);
+        devc->packet_len = 0;
+    }
+
+    len = sizeof(devc->packet) - devc->packet_len;
+    data = &devc->packet[devc->packet_len];
+    len = serial_read_nonblocking(serial,
+                                  data, len);
+    if (!len)
+        return 0;
+
+    devc->packet_len += len;
+    process_buffer(sdi);
+
+    return 0;
+}
+
+SR_PRIV int ut8802e_handle_events(int fd, int revents, void *cb_data) {
+    struct sr_dev_inst *sdi;
+    struct sr_serial_dev_inst *serial;
+
+    (void)fd;
+
+    sdi = cb_data;
+    if (!sdi)
+        return TRUE;
+    serial = sdi->conn;
+    if (!serial)
+        return TRUE;
+
+    if (revents & G_IO_IN)
+        (void)ut8802e_receive_data(sdi);
```

Comment:
If you don't want to know if that function returned an error, just rewrite the function to not return one.